lib/airbrake/rack/instrumentable.rb

```diff
@@ -15,12 +17,99 @@ module Rack
     # @since v9.2.0
     module Instrumentable
       def airbrake_capture_timing(method_name, label: method_name.to_s)
-        alias_method "#{method_name}_without_airbrake", method_name
+        mod = ::Airbrake::Rack::Instrumentable
+        if mod.should_prepend?(self, method_name)
+          mod.prepend_airbrake_capture_timing(self, method_name, label)
+        else
+          mod.chain_airbrake_capture_timing(self, method_name, label)
+        end
+        method_name
+      end
+
+      # @api private
+      def __airbrake_capture_timing_module__
+        # Module used to store prepended wrapper methods, saved as an instance
+        # variable so each target class/module gets its own module. This just
+        # a convenience to avoid prepending a lot of anonymous modules.
+        @__airbrake_capture_timing_module__ ||= ::Module.new
+      end
+
+      # Using api private self methods so they don't get defined in the target
+      # class or module, but can still be called by the above method.
+
+      # @api private
+      def self.should_prepend?(klass, method_name)
+        # Don't chain already-prepended or operator methods.
+        klass.module_exec do
+          self_class_idx = ancestors.index(self)
+          method_owner_idx = ancestors.index(instance_method(method_name).owner)
+          method_owner_idx < self_class_idx || !(/\A\W/ =~ method_name).nil?
+        end
+      end
 
-        define_method(method_name) do |*args|
-          Airbrake::Rack.capture_timing(label) do
-            __send__("#{method_name}_without_airbrake", *args)
+      # @api private
+      def self.prepend_airbrake_capture_timing(klass, method_name, label)
+        args = method_signature
+        klass.module_exec do
+          # Check the visibility.
+          is_protected = protected_method_defined?(method_name)
+          is_private = private_method_defined?(method_name)
+          mod = __airbrake_capture_timing_module__
+          mod.module_exec do
+            # Generate the wrapper method.
+            module_eval <<-RUBY, __FILE__, __LINE__ + 1
+              def #{method_name}(#{args})
+                Airbrake::Rack.capture_timing(#{label.to_s.inspect}) do
```

Comment:
We need to call `.inspect` because it adds the quotes around the string, because this is interpolating into source code. We call the `.to_s` because I don't want to make too many assumptions about what label is, and string inspect definitely generates source compatible values. On 2.7+, `.to_s.inspect` even returns the same frozen string every time for symbols. We can get rid of the `.to_s` safely for values of `label` that the code supports, if you like, but we need the `.inspect`.